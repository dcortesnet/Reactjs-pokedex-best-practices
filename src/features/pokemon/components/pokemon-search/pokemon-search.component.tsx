import React, { useContext, useState } from "react";
import * as pokemonService from "../../../../shared/services/pokemon.service";
import { PokemonContext } from "../../../../shared/context/pokemon.context";
import { PokemonSearchLayout } from "./pokemon-search.layout";

export const PokemonSearchComponent = () => {
  const [text, setText] = useState("");
  const { setPokemon, setNextPage } = useContext(PokemonContext);

  const handleChange = (e: any) => {
    setText(e.target.value.toLowerCase());
  };

  const searchPokemon = async (e: any) => {
    e.preventDefault();
    if (text) {
      const pokemon = await pokemonService.getPokemonByName(text);
      setPokemon([pokemon]);
    } else {
      const page = await pokemonService.getPokemonPage();
      const pokemon = await pokemonService.getAllPokemonByPage(page);
      setNextPage(page.next);
      setPokemon(pokemon);
    }
  };

  return (
    <PokemonSearchLayout
      onChange={handleChange}
      onSubmit={searchPokemon}
    ></PokemonSearchLayout>
  );
};
