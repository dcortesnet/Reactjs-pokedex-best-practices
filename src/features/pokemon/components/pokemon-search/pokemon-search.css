.navbar {
  width: 100%;
  background: #f00000;
}

.navbar-container {
  padding-block-start: 20px;
  padding-block-end: 20px;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.logo {
  display: block;
}

.searchBar {
  padding-block-start: 20px;
  border: 1px solid transparent;
}

form {
  display: flex;
  flex-direction: column;
  gap: 2px;
  width: 100%;
}

input {
  padding: 10px;
  margin-left: 20px;
  margin-right: 20px;
}

button {
  padding: 10px;
  cursor: pointer;
  margin-left: 20px;
  margin-right: 20px;
}

@media screen and (min-width: 460px) {
  form {
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }

  input {
    width: 20%;
    margin: 0;
  }

  button {
    margin: 0;
  }
}
