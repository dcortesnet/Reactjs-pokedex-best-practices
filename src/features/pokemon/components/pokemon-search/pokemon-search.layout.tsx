import pokeApiLogo from "../../../../shared/assets/poke_api_logo.png";
import "./pokemon-search.css";

interface Props {
  onChange: Function;
  onSubmit: Function;
}

export const PokemonSearchLayout: React.FC<Props> = ({
  onChange,
  onSubmit,
}) => {
  return (
    <nav className="navbar">
      <div className="navbar-container">
        <img className="logo" src={pokeApiLogo} alt="pokeApiLogo" />
        <form onSubmit={(e) => onSubmit(e)} className="searchBar">
          <input
            name="search"
            onChange={(e) => onChange(e)}
            placeholder="Buscador por nombre o nÃºmero"
          ></input>
          <button type="submit">Buscar</button>
        </form>
      </div>
    </nav>
  );
};
