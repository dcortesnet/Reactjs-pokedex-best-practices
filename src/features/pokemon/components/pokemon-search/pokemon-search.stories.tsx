import { Story, Meta } from "@storybook/react";
import { PokemonSearchLayout } from "./pokemon-search.layout";

export default {
  title: "Pokemon/components/PokemonSearch",
  component: PokemonSearchLayout,
} as Meta;

export const Template: Story<any> = (args) => <PokemonSearchLayout {...args}/>;
