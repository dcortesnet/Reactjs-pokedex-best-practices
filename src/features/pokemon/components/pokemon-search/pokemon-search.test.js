import { PokemonSearchLayout } from "./pokemon-search.layout";
import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("PokemonSearchLayout", () => {
  it("Renders correctly", () => {
    const r = shallow(<PokemonSearchLayout></PokemonSearchLayout>);

    expect(toJson(r)).toMatchSnapshot();
  });


  it("Renders and has 'Buscar' text", () => {
    const text = "Buscar";
    const r = shallow(<PokemonSearchLayout></PokemonSearchLayout>);

    expect(r.contains(text)).toEqual(true);
  });
});
