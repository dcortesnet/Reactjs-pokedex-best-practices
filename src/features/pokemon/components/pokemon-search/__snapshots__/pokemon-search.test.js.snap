// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonSearchLayout Renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-container"
  >
    <img
      alt="pokeApiLogo"
      className="logo"
      src="poke_api_logo.png"
    />
    <form
      className="searchBar"
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="Buscador por nombre o nÃºmero"
      />
      <button
        type="submit"
      >
        Buscar
      </button>
    </form>
  </div>
</nav>
`;
